tests/lib/rules/await-async-utils.test.ts

```diff
@@ -258,6 +258,40 @@ ruleTester.run(RULE_NAME, rule, {
         test('edge case for no innermost function scope', () => {
           const foo = waitFor
         })
+      `,
+		},
+		{
+			code: `
+        function setup() {
```

Comment:
Please thoroughly review the test cases any let me know if some are incorrectly marked as valid and vice versa :) 

Also:  Should I make use of the `...ASYNC_UTILS.map()` pattern used in other test cases in my new test cases as well?
*

Reply:
Yes, you should wrap your test cases in `...ASYNC_UTILS.map()` so all async utils are automatically tested.

Reply:
@Belco90, I've changed the test cases in the latest commit by
1. Using `...ASYNC_UTILS.map()`
2. Renaming `waitForElementToBeRemoved` to `waitForAsyncUtil` inside the test cases (just a small change in order to align it better)

Feel free to have another look :)